feat: Updates documentation with information on updating wit deps

thomastaylor312 opened the pull request:
This attempts to find any place that references the old `wit-deps` method and remove that information. It also adds a new subsection to the developer docs section on building to explain the automatic dependency fetching

thomastaylor312 committed:
feat: Updates documentation with information on updating wit deps

This attempts to find any place that references the old `wit-deps`
method and remove that information. It also adds a new subsection to
the developer docs section on building to explain the automatic
dependency fetching

Signed-off-by: Taylor Thomas <taylor@cosmonic.com>

ericgregory commented on docs/cli/index.mdx:
```diff
@@ -1,8 +1,3 @@
-sidebar_position: 9
----
-
 # Command-Line Help for `wash`
```
```suggestion
---
title: 'CLI (wash)'
sidebar_position: 9
---

# Command-Line Help for `wash`
```

thomastaylor312 replied:
Oh whoops. Yay for autogeneration

thomastaylor312 committed:
feat: Updates wasmcloud.toml with new options

This documents the new options available for specifying a wit directory
and overriding dependencies

Signed-off-by: Taylor Thomas <taylor@cosmonic.com>

ericgregory commented on docs/developer/providers/build.md:
```diff
@@ -79,3 +79,53 @@ x86_64-macos
+## Interface dependencies and `wash build`
+
+:::warning
+Some provider implementations may have issues with the new dependency fetching see [the section below on migration](#migrating-from-wit-deps) for additional details
```
```suggestion
Some provider implementations may have issues with the new dependency fetching&mdash;see [the section below on migration](#migrating-from-wit-deps) for additional details.
```
